src/invstorage.cc

```diff
@@ -52,9 +52,63 @@ void INvStorage::importData(const ImportConfig& import_config) {
 boost::shared_ptr<INvStorage> INvStorage::newStorage(const StorageConfig& config) {
   switch (config.type) {
     case kSqlite:
+      if (!boost::filesystem::exists(config.sqldb_path)) {
+        boost::shared_ptr<INvStorage> sql_storage(new SQLStorage(config));
+        boost::shared_ptr<INvStorage> fs_storage(new FSStorage(config));
+        INvStorage::FSSToSQLS(fs_storage, sql_storage);
+        return sql_storage;
+      }
       return boost::shared_ptr<INvStorage>(new SQLStorage(config));
-    case kFile:
+    case kFileSystem:
     default:
       return boost::shared_ptr<INvStorage>(new FSStorage(config));
   }
+}
+
+void INvStorage::FSSToSQLS(const boost::shared_ptr<INvStorage>& fs_storage,
+                           boost::shared_ptr<INvStorage>& sql_storage) {
```

Comment:
Reference-to-pointer is probably unnecessary. This came up recently for some of my code as well and I'm not sure there's a good reason for the reference.

Reply:
It is reference to object(instance of shared_ptr) that holds pointer, not to pointer itself.  So I don't see any reason to make a copy of that instance. 
https://stackoverflow.com/questions/3310737/shared-ptr-by-reference-or-by-value

Reply:
Good point. Seems fine to me upon reconsideration.